Lib/os.py

```diff
@@ -651,6 +651,29 @@ def get_exec_path(env=None):
     return path_list.split(pathsep)
 
 
+def get_current_dir_name():
+    """
+    Return a string representing the current working directory taking into
+    consideration the users *PWD* environment variable if it exists. This
+    is opposed to getcwd() which dereferences symlinks in the path. This
+    function is identical to getcwd() on systems that do *not* support
+    the *PWD* environment variable.
+    """
+    cwd = getcwd()
+
+    if name == 'nt':
```

Comment:
Why this is needed?

Reply:
I think you were performing review as I was writing a comment :)
https://github.com/python/cpython/pull/10117#discussion_r228702427

This function is documented with:
> This function is identical to :func:`getcwd` on systems that do **not** support the :envvar:`PWD` environment variable.

Please correct me if I'm wrong since I'm less familiar with Windows, but I don't believe it uses the `PWD` environment variable in any way. I added that check so that we would not incorrectly change behavior from `getcwd()` if the user does set the `PWD` on Windows for some reason.

Reply:
in `git bash` on windows:

```
Anthony@AnthonysDesktop MINGW64 ~
$ python -c 'import os; print(os.getenv("PWD"))'
C:/Users/Anthony
```

Reply:
@asottile
Thanks. Looks like my memory was incorrect. I'll fix this up.

Reply:
I'm actually not sure this function is useful/a good idea -- it's entirely dependent on whether `python` is executed in the context of an interactive shell that sets or doesn't set `PWD`.

I've commented on the bpo issue indicating that as well.